### PR TITLE
Eliminate wrapped enum

### PR DESCRIPTION
**Pull request by diyessi**
Switch allreduce to new op form

### CODE REVIEW

File: src/ngraph/distributed.hpp [hunk 1]
```diff
@@ -26,34 +26,15 @@ namespace ngraph
 {
     namespace reduction
     {
-        enum class Type_t
+        enum class Type
         {
             sum,
```
Comment (rkimballn1) [inline src/ngraph/distributed.hpp]: Upper case enum elements

File: src/ngraph/op/allreduce.hpp [hunk 2]
```diff
@@ -29,17 +29,17 @@ namespace ngraph
             NGRAPH_API
             static const std::string type_name;
             const std::string& description() const override { return type_name; }
-            AllReduce();
-            AllReduce(const std::shared_ptr<Node>& arg,
-                      const reduction::Type reduce_type = reduction::sum);
+            AllReduce() = default;
+            AllReduce(const Output<Node>& arg, reduction::Type reduce_type = reduction::Type::sum);
```
Comment (jianyinglang) [inline src/ngraph/op/allreduce.hpp]: Should the `sum` here should be capital case too?

File: src/ngraph/op/allreduce.hpp [hunk 3]
```diff
@@ -29,17 +29,17 @@ namespace ngraph
...
 
             void validate_and_infer_types() override;
 
             std::shared_ptr<Node> copy_with_new_args(const NodeVector& new_args) const override;
             reduction::Type get_reduce_type() const;
+            void set_reduce_type(reduction::Type reduce_type);
 
         private:
-            const reduction::Type m_reduce_type;
+            reduction::Type m_reduce_type{reduction::Type::sum};
```
Comment (jianyinglang) [inline src/ngraph/op/allreduce.hpp]: Same here

File: test/distributed.in.cpp [hunk 4]
```diff
@@ -50,25 +50,25 @@ static void test_allreduce_common(reduction::Type reduce_type)
 #pragma GCC diagnostic error "-Wswitch"
 #pragma GCC diagnostic error "-Wswitch-enum"
 #endif
-        switch (reduce_type.get_type())
+        switch (reduce_type)
         {
-        case reduction::Type_t::sum:
+        case reduction::Type::sum:
             copy_data(a, v);
```
Comment (jianyinglang) [inline test/distributed.in.cpp]: Capital case?

**Pull request merged**